ERCS/erc-7679.md

```diff
@@ -0,0 +1,212 @@
+---
+eip: 7679
+title: UserOperation Builder
+description: Construct UserOperations without being coupled with account-specific logic.
+author: Derek Chiang (@derekchiang), Garvit Khatri (@plusminushalf), Fil Makarov (@filmakarov), Kristof Gazso (@kristofgazso), Derek Rein (@arein), Tomas Rocchi (@tomiir), bumblefudge (@bumblefudge)
+discussions-to: https://ethereum-magicians.org/t/erc-7679-smart-account-interfaces/19547
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-05
+requires: 4337
+---
+
+## Abstract
+
+Different [ERC-4337](./eip-4337.md) smart account implementations encode their signature, nonce, and calldata differently.  This makes it difficult for DApps, wallets, and smart account toolings to integrate with smart accounts without integrating with account-specific SDKs, which introduces vendor lock-in and hurts smart account adoption.
+
+We propose a standard way for smart account implementations to put their account-specific encoding logic on-chain.
```

Comment:
Could you include a bit more technical detail on how you put the encoding logic on chain? Just a sentence or two to give the reader a high level overview.